pallets/drand/src/lib.rs

```diff
@@ -502,13 +503,21 @@ impl<T: Config> Pallet<T> {
 	}
 
 	/// get the randomness at a specific block height
-	/// returns [0u8;32] if it does not exist
-	pub fn random_at(block_number: BlockNumberFor<T>) -> [u8; 32] {
-		let pulse = Pulses::<T>::get(block_number).unwrap_or_default();
-		let rand = pulse.randomness.clone();
-		let bounded_rand: [u8; 32] = rand.into_inner().try_into().unwrap_or([0u8; 32]);
-
-		bounded_rand
+	/// returns None if it is invalid or does not exist
+	pub fn random_at(block_number: BlockNumberFor<T>) -> Option<RandomValue> {
+		match Pulses::<T>::get(block_number) {
+			Some(pulse) => {
+				let rand = pulse.randomness.into_inner();
```

Comment:
What if we did: `let bounded_rand: [u8; 32] = rand.into_inner().try_into().ok()`? then we don't need to explicitly comment on the 'should never happen' branch.

Reply:
definitely a much nicer solution, applied